List/LocationList.py

```diff
@@ -441,21 +441,21 @@ def getTreasureList():
-            KH2Treasure(538, "Musical Orichalcum+", LocationTypes=[locationType.Atlantica],  InvalidChecks=[itemType.MAGNET, itemType.THUNDER, itemType.TORN_PAGE, itemType.FORM, itemType.GROWTH_ABILITY, itemType.SUPPORT_ABILITY, itemType.ACTION_ABILITY]),
+            KH2Treasure(385, "Ruin and Creation's Passage Mythril Stone", LocationTypes=[locationType.TWTNW]).setLocationWeight(10),
+            KH2Treasure(386, "Ruin and Creation's Passage AP Boost", LocationTypes=[locationType.TWTNW]).setLocationWeight(10),
+            KH2Treasure(387, "Ruin and Creation's Passage Mythril Crystal", LocationTypes=[locationType.TWTNW]).setLocationWeight(10),
+            KH2Treasure(388, "Ruin and Creation's Passage Orichalcum", LocationTypes=[locationType.TWTNW]).setLocationWeight(10),
+            KH2Treasure(537, "Xemnas 1 Secret Ansem Report 13", LocationTypes=[locationType.TWTNW], InvalidChecks=[itemType.GROWTH_ABILITY, itemType.SUPPORT_ABILITY, itemType.ACTION_ABILITY]).setLocationWeight(10),
+            KH2Treasure(554, "Xemnas (Data) Power Boost", LocationTypes=[locationType.TWTNW, locationType.DataOrg], InvalidChecks=[itemType.GROWTH_ABILITY, itemType.SUPPORT_ABILITY, itemType.ACTION_ABILITY]).setLocationWeight(10),
+
+            KH2Treasure(585, "Garden of Assemblage Map", LocationTypes=[locationType.Free]).setLocationWeight(-10),
+            KH2Treasure(586, "GoA Lost Illusion", LocationTypes=[locationType.Free]).setLocationWeight(-10),
+            KH2Treasure(590, "Proof of Nonexistence", LocationTypes=[locationType.Free]).setLocationWeight(-10),
+
+            KH2Treasure(367, "Undersea Kingdom Map", LocationTypes=[locationType.Atlantica], InvalidChecks=([] if trueRandom else [itemType.MAGNET, itemType.THUNDER, itemType.TORN_PAGE, itemType.FORM])+ [itemType.GROWTH_ABILITY, itemType.SUPPORT_ABILITY, itemType.ACTION_ABILITY]),
+            KH2Treasure(287, "Mysterious Abyss", LocationTypes=[locationType.Atlantica], InvalidChecks=([] if trueRandom else [itemType.MAGNET, itemType.THUNDER, itemType.TORN_PAGE, itemType.FORM]) + [itemType.GROWTH_ABILITY, itemType.SUPPORT_ABILITY, itemType.ACTION_ABILITY]).setLocationWeight(10),
+            KH2Treasure(279, "Musical Blizzard Element", LocationTypes=[locationType.Atlantica],  InvalidChecks=([] if trueRandom else [itemType.MAGNET, itemType.THUNDER, itemType.TORN_PAGE, itemType.FORM]) + [itemType.GROWTH_ABILITY, itemType.SUPPORT_ABILITY, itemType.ACTION_ABILITY]).setLocationWeight(10),
+            KH2Treasure(538, "Musical Orichalcum+", LocationTypes=[locationType.Atlantica],  InvalidChecks=([] if trueRandom else [itemType.MAGNET, itemType.THUNDER, itemType.TORN_PAGE, itemType.FORM]) + [itemType.GROWTH_ABILITY, itemType.SUPPORT_ABILITY, itemType.ACTION_ABILITY]).setLocationWeight(10),
 ]
 
     def getSoraLevelList():
```

Comment:
Levels probably deserve some weighting, levels 1-15 early, and 40+ late?

Reply:
I wasn't sure if encouraging "hard" settings to have late levels was good, since that makes it less hard. But now I think that making checks late in levels is just time consuming, and I would prefer not adding weight on levels for higher difficulties. I agree for easier difficulties adding some weight in the early levels though.